File: VeamIt-project/iOS/veam00000000/ConsoleUpdatePreparingMixedStatusHandler.m
//
//  ConsoleUpdatePreparingMixedStatusHandler.m
//  veam00000000
//
//  Created by veam on 1/6/15.
//  Copyright (c) 2015 veam. All rights reserved.
//

#import "ConsoleUpdatePreparingMixedStatusHandler.h"
#import "ConsoleUtil.h"

@implementation ConsoleUpdatePreparingMixedStatusHandler

- (void)handlePostResult:(NSArray *)results
{
    //NSLog(@"%@::handlePostResult",NSStringFromClass([self class])) ;
    if([results count] >= 2){
        //NSLog(@"count >= 2") ;
        NSString *result = [results objectAtIndex:0] ;
        if([result isEqualToString:@"OK"]){
            ConsoleContents *contents = [ConsoleUtil getConsoleContents] ;
            NSString *elementString = [results objectAtIndex:1] ;
            //NSLog(@"elementString=%@",elementString) ;
            NSArray *elements = [elementString componentsSeparatedByString:@","] ;
            int numberOfElements = [elements count] ;
            for(int index = 0 ; index < numberOfElements ; index++){
                NSString *valueString = [elements objectAtIndex:index] ;
                NSArray *values = [valueString componentsSeparatedByString:@"|"] ;
                if([values count] >= 4){
                    NSString *mixedId = [values objectAtIndex:0] ;
                    NSString *status = [values objectAtIndex:1] ;
                    NSString *statusText = [values objectAtIndex:2] ;
                    NSString *thumbnailUrl = [values objectAtIndex:3] ;
                    Mixed *mixed = [contents getMixedForId:mixedId] ;
                    if(mixed != nil){
                        [mixed setStatus:status] ;
                        [mixed setStatusText:statusText] ;
                        [mixed setThumbnailUrl:thumbnailUrl] ;
                    }
                }
            }
        }
    }
}

@end

File: VeamIt-project/Android/veam31000287/veam31000287/src/main/java/co/veam/veam31000287/ConsoleUpdatePreparingMixedStatusHandler.java
package co.veam.veam31000287;

import android.util.Log;

import java.util.ArrayList;

/**
 * Created by veam on 11/8/16.
 */
public class ConsoleUpdatePreparingMixedStatusHandler implements HandlePostResultInterface {
    @Override
    public void handlePostResult(ArrayList<String> results) {
        VeamUtil.log("debug", "ConsoleUpdatePreparingMixedStatusHandler::handlePostResult") ;
        if(results.size()  >= 2){
            VeamUtil.log("debug", "count >= 2") ;
            String result = results.get(0) ;
            if(result.equals("OK")){
                ConsoleContents contents = ConsoleUtil.getConsoleContents() ;
                String elementString = results.get(1) ;
                VeamUtil.log("debug", "elementString="+elementString) ;
                String[] elements = elementString.split(",") ;
                int numberOfElements = elements.length ;
                for(int index = 0 ; index < numberOfElements ; index++){
                    String valueString = elements[index] ;
                    VeamUtil.log("debug", "valueString="+valueString) ;
                    String[] values = valueString.split("\\|") ;
                    if(values.length >= 4){
                        String mixedId = values[0] ;
                        String status = values[1] ;
                        String statusText = values[2] ;
                        String thumbnailUrl = values[3] ;
                        VeamUtil.log("debug", "mixedId="+mixedId) ;
                        VeamUtil.log("debug", "status="+status) ;
                        VeamUtil.log("debug", "statusText="+statusText) ;
                        MixedObject mixed = contents.getMixedForId(mixedId) ;
                        if(mixed != null){
                            mixed.setStatus(status) ;
                            mixed.setStatusText(statusText) ;
                            mixed.setThumbnailUrl(thumbnailUrl) ;
                            VeamUtil.log("debug", "set id=" + mixed.getId() + " status=" + mixed.getStatus()) ;
                        }
                    }
                }
            }
        }
    }
}
